testsrc/org/mozilla/javascript/tests/NativeJavaListTest.java

```diff
@@ -88,6 +88,263 @@ public void testUpdatingJavaListStringValues() {
+    }
+    
+    public void testDelete() {
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+//        Object list = runScript("['a','b','c']", null, Function.identity());
+        // TODO: should NativeJavaList distinguish between 'null' and 'undefined'?
+        assertEquals("false", runScriptAsString("delete value[1]", list));
+        assertEquals("a,,c", runScriptAsString("value.join()", list));
+    }
+
+    public void testArrayConcat() {
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+        assertEquals("x,a,b,c", runScriptAsString("['x'].concat(value).join()", list));
+        assertEquals("a,b,c,x", runScriptAsString("value.concat(['x']).join()", list));
+    }
+    
+    public void testArrayCopyWithin() {
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+        list.add("d");
+        list.add("e");
+        // copy to index 0 the element at index 3
+        assertEquals("d,b,c,d,e", runScriptAsString("value.copyWithin(0, 3, 4).join()", list));
+    }
+
+    public void testArrayFill() {
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+        assertEquals("a,d,d", runScriptAsString("value.fill('d', 1, 4).join()", list));
+        assertEquals("[a, d, d]", list.toString());
+    }
+
+    public void testArrayIncludes() {
+
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+        assertEquals("true", runScriptAsString("value.includes('b')", list));
+        assertEquals("false", runScriptAsString("value.includes('d')", list));
+
+        List<Double> listD = new ArrayList<>();
+        listD.add(1.0);
+        listD.add(2.0);
+        listD.add(3.0);
+        assertEquals("true", runScriptAsString("value.includes(2)", listD));
+        assertEquals("false", runScriptAsString("value.includes(4)", listD));
+        
+        List<Integer> listI = new ArrayList<>();
+        listI.add(1);
+        listI.add(2);
+        listI.add(3);
+        assertEquals("true", runScriptAsString("value.includes(2)", listI));
+        assertEquals("false", runScriptAsString("value.includes(4)", listD));
+    }
+    
+    public void testArrayIndexOf() {
+        
+        List<String> list = new ArrayList<>();
+        list.add("a");
+        list.add("b");
+        list.add("c");
+        assertEquals("1", runScriptAsString("value.indexOf('b')", list));
+        assertEquals("-1", runScriptAsString("value.indexOf('d')", list));
+        
+        List<Double> listD = new ArrayList<>();
+        listD.add(1.0);
+        listD.add(2.0);
+        listD.add(3.0);
+        assertEquals("1", runScriptAsString("value.indexOf(2)", listD));
```

Comment:
Isn't this calling Java's `indexOf` method?

https://docs.oracle.com/javase/8/docs/api/java/util/List.html#indexOf-java.lang.Object-
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf

They differ in the following ways.
- JS can receive `fromIndex` as the second argument.
```js
var list = new java.util.ArrayList();
list.indexOf(1, 2);
// js: Can't find method java.util.ArrayList.indexOf(number,number).
```
- The equivalence check is different.
```js
list.add(NaN);
list.indexOf(NaN); // 0

[NaN].indexOf(NaN); // -1
```
https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#equals-java.lang.Object-

There may be compatibility issues.